### PR TITLE
Show use of @Trace annotation and thread propagation in Java agent

### PR DESCRIPTION
**Pull request by keitwb**

### CODE REVIEW

File: java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java [hunk 1]
```diff
@@ -1,89 +1,120 @@
...
+            }
 
-    Jedis redisClient = new Jedis("localhost");
+            setInRedisWithThread(url, respBody);
+        } catch (Throwable e) {
+            Tags.ERROR.set(span, true);
+            span.log(Collections.singletonMap(ERROR_OBJECT, e));
+            System.out.println(e);
+        } finally {
+            span.finish();
```
Comment (rmfitzpatrick) [inline java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java]: Is there a risk for this root span finishing before the traced redis SET's does?

Comment (keitwb) [reply java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java]: Yes, but I don't know of anything in OT that requires spans to be temporally nested, it only defines child spans as causally related.  Their [span timing diagram](https://github.com/opentracing/specification/blob/master/specification.md#references-between-spans) even shows child spans extending past the time of parent spans.  Our backend doesn't have any issues with it either from what I have seen.  Do you know of anything bad that could happen in such a scenario?

Comment (mpetazzoni) [reply java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java]: Nope that's perfectly fine, that's how async processes work:

```
[ op-that-schedules-an-async-task ]
  +------[ a-much-longer-async-task-that-may-finish-much-later-than-its-parent ]
```

File: java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java [hunk 2]
```diff
@@ -1,89 +1,120 @@
...
+        } catch (InterruptedException e) {
+            return;
+        }
     }
 
-    redisClient.close();
+    /*
+     * This annotation will generate a span that wraps the auto-generated Redis SET command span that is
+     * run within a separate thread.  This is a fairly
+     * trivial example that doesn't add much value but at least shows the use of the Trace annotation.
```
Comment (mpetazzoni) [inline java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java]: Might be good to show accessing the current span (that the `@Trace` generates) to add metadata tags to it in the method's body.

**Pull request merged**